sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/CosmosCatalog.scala

```diff
@@ -290,26 +339,77 @@ class CosmosCatalog
         .block()
     }
 
+    val effectiveOptions = tableOptions ++ containerProperties
+
     new ItemsTable(
       sparkSession,
       partitions,
       Some(databaseName),
       Some(containerName),
-      tableOptions.asJava,
+      effectiveOptions.asJava,
       Option.apply(schema))
   }
 
-  @throws(classOf[UnsupportedOperationException])
-  override def alterTable(ident: Identifier, changes: TableChange*): Table = {
-    // TODO: moderakh we can support updating indexing policy and throughput
-    throw new UnsupportedOperationException
+  //scalastyle:off method.length
+  @tailrec
+  private def createViewTable(ident: Identifier,
+                              databaseName: String,
+                              viewName: String,
+                              schema: StructType,
+                              partitions: Array[Transform],
+                              containerProperties: Map[String, String]): Table = {
+
+    logInfo(s"createViewTable DB:$databaseName, View: $viewName")
+
+    this.viewRepository match {
+      case Some(viewRepositorySnapshot) =>
+        val userProvidedSchema = if (schema != null && schema.length > 0) {
+          Some(schema)
+        } else {
+          None
+        }
+        val viewDefinition = ViewDefinition(databaseName, viewName, userProvidedSchema, containerProperties)
+        var lastBatchId = 0L
+        val newViewDefinitionsSnapshot = viewRepositorySnapshot.getLatest() match {
+          case Some(viewDefinitionsSnapshot) =>
+            lastBatchId = viewDefinitionsSnapshot._1
+            val alreadyExistingViews = viewDefinitionsSnapshot._2
+
+            if (alreadyExistingViews.exists(v => v.databaseName.equalsIgnoreCase(databaseName) &&
```

Comment:
cosmos db "ids" and database name is case sensitive.

any reason we are using equalsIgnoreCase on the databasename

Reply:
No - it was a bug. Thanks for catching it.